landing-page/src/components/landing-page-network.ts

```diff
@@ -0,0 +1,163 @@
+
+  @state() private _dnsPrimaryInterface?: string;
+
+  protected render() {
+    if (!this._networkIssue && !this._getNetworkInfoError) {
+      return nothing;
+    }
+
+    if (this._getNetworkInfoError) {
+      return html`
+        <ha-alert alert-type="error">
+          <p>${this.localize("network_issue.error_get_network_info")}</p>
+        </ha-alert>
+      `;
+    }
+
+    return html`
+      <ha-alert
+        alert-type="warning"
+        .title=${this.localize("network_issue.title")}
+      >
+        <p>
+          ${this.localize("network_issue.description", {
+            dns: this._dnsPrimaryInterface || "?",
+          })}
+        </p>
+        <p>${this.localize("network_issue.resolve_different")}</p>
+        ${!this._dnsPrimaryInterface
+          ? html`
+              <p>
+                <b>${this.localize("network_issue.no_primary_interface")} </b>
+              </p>
+            `
+          : nothing}
+        <div class="actions">
+          ${ALTERNATIVE_DNS_SERVERS.map(
+            ({ translationKey }, key) =>
+              html`<ha-button
+                .index=${key}
+                .disabled=${!this._dnsPrimaryInterface}
+                @click=${this._setDns}
+                >${this.localize(translationKey)}</ha-button
+              >`
+          )}
+        </div>
+      </ha-alert>
+    `;
+  }
+
+  protected firstUpdated(_changedProperties: PropertyValues): void {
+    super.firstUpdated(_changedProperties);
+    this._fetchSupervisorInfo();
+  }
+
+  private _scheduleFetchSupervisorInfo() {
+    setTimeout(
+      () => this._fetchSupervisorInfo(),
+      SCHEDULE_FETCH_NETWORK_INFO_SECONDS * 1000
+    );
+  }
+
+  private async _fetchSupervisorInfo() {
+    try {
+      const response = await getSupervisorNetworkInfo();
+      if (!response.ok) {
+        throw new Error("Failed to fetch network info");
+      }
+
+      const { data } = await response.json();
+
+      this._getNetworkInfoError = false;
+
+      if (!data.host_internet) {
+        this._networkIssue = true;
+        const primaryInterface = data.interfaces.find(
+          (intf) => intf.primary && intf.enabled
+        );
+        if (primaryInterface) {
+          this._dnsPrimaryInterface = [
+            ...(primaryInterface.ipv4?.nameservers || []),
+            ...(primaryInterface.ipv6?.nameservers || []),
+          ].join(", ");
+        }
+      } else {
+        this._networkIssue = false;
+      }
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.error(err);
+      this._getNetworkInfoError = true;
+    }
+
+    fireEvent(this, "value-changed", {
+      value: this._networkIssue || this._getNetworkInfoError,
+    });
+    this._scheduleFetchSupervisorInfo();
+  }
+
+  private async _setDns(ev) {
+    const index = ev.target?.index;
+    try {
+      const response = await setSupervisorNetworkDns(index);
+      if (!response.ok) {
+        throw new Error("Failed to set DNS");
```

Comment:
Same, we don't handle this error

Reply:
It is inside a try block. But it handles the same as network info error, I can add an additonal error info for failed set dns.

Reply:
This one is more important I think, as it is an action a user performs, we now show a message cannot get network info when they try to set the dns severs.